### ISSUE TITLE
Blurry 24px action icons in Qt/GTK2 filechoosers and Octave, VLC

### ISSUE THREAD

**Issue by palob** [numixproject/numix-icon-theme#948]
![blurry1](https://cloud.githubusercontent.com/assets/7050624/11766941/578a0e90-a19c-11e5-80fc-589491027c50.png)
![blurry2](https://cloud.githubusercontent.com/assets/7050624/11766942/578c5aa6-a19c-11e5-8a00-a9c6c3c14821.png)
![blurry3](https://cloud.githubusercontent.com/assets/7050624/11766940/57815a84-a19c-11e5-9875-5aacf0e8b636.png)

The issue is due to 16px 22px or 32px (don't know exactly) action icons showing up in place of 24px ones.

This issue isn't present with other themes like Adwaita, breeze or elementary.

Adwaita and breeze have set ``Type=Fixed`` for actions icons in all sizes.
elementary has set ``Type=Scalabe`` like Numix (and the same Min/Max sizes except for 24px) but they haven't got 22px action icons.

Setting ``Type=Fixed`` in the Numix ``index.theme`` for 16,22,24,32 actions and the respective @x2 ones (haven't pinpointed it any more specifically) fixes the issue.

What do we need ``Type=Scalable`` for anyway?
Are they any non-hardcoded action icons with other sizes than 16,22,24,32....?
Those would show up blurry anyway.

**Comment by wa4557** [comment]
We actually don't need the `scalable` for smaller sizes (since Numix is more or less complete in all sizes). Only for 64px and 48px the scalable is necessary because they are used for 96px/128px and 256px

**Comment by palob** [comment]
So we don't need it for status, mimetypes etc. either?

**Comment by wa4557** [comment]
No not really. Its only needed in all all `64px` (except `places` where it can be removed as well) all `256px` and `48/notifications`

**Comment by wa4557** [comment]
But I'm wondering why this is happening because there is the `MaxSize=23` which should prevent `22px` icons used instead of `24px` ones

**Comment by palob** [comment]
Maybe it's 16px or 32px showing up?

**Comment by wa4557** [comment]
Maybe it's the 32px icon. Can you try setting the `MinSize` variable for the `32px` icons to `32`?

**Comment by palob** [comment]
That didn't fix it.

**Comment by wa4557** [comment]
Ok. Anyways I think we can make them `Type=fixed` again. I think it might be Qt using a different icon lookup mechanism as GTK

**Comment by wa4557** [comment]
Btw: Where do you have Numix icons in VLC?

**Comment by palob** [comment]
In the main GUI view there's ``view-list`` to choose how to display the playlist. 
When opening a file in GNOME Shell it uses the legacy GTK2 filechooser showing ``gtk-edit`` blurry.

**Comment by wa4557** [comment]
I can confirm that. It is indeed blurry...

**Comment by palob** [comment]
In breeze there are 8px emblems so I take it I better leave 16px emblems at ``Scalable``?

**Comment by wa4557** [comment]
True, yes.

**Comment by hlechner** [comment]
I have checked into the vlc's source code and made some extra tests.

If you go to **Tools** > **Preference** and change the **Force window style** and select anything else instead of **System default**, **GTK+** or **Cleanlooks** you'll check that the icon will show exactly 16px not blurry.

My tests are directed to the "changeplaylistview", the button that show on bar.

Here the important file:
``modules/gui/qt4/components/playlist/playlist.cpp``

A bit of the code:
```
    /* Button to switch views */
    QToolButton *viewButton = new QToolButton( this );
    viewButton->setIcon( style()->standardIcon( QStyle::SP_FileDialogDetailedView ) );
    viewButton->setToolTip( qtr("Change playlistview") );
    topbarLayout->addWidget( viewButton );

    viewButton->setMenu( StandardPLPanel::viewSelectionMenu( mainView ));
    CONNECT( viewButton, clicked(), mainView, cycleViews() );
```

Basically it uses **QStyle** function of QT, the specific icon is defined by [**SP_FileDialogDetailedView**](https://doc.qt.io/archives/qtjambi-4.5.2_01/com/trolltech/qt/gui/QStyle.StandardPixmap.html#SP_FileDialogDetailedView).

I also found some references about the icon size depending on the bar size itself, that is defined on GTK theme:
![image](https://cloud.githubusercontent.com/assets/11244067/11768486/49a034a8-a1b5-11e5-90d7-03f3bcc40d58.png)
The GTK+ theme above is called Ceti-2 as you can see is very different from adwaita and other themes:
![image](https://cloud.githubusercontent.com/assets/11244067/11768490/8098fa4e-a1b5-11e5-9b83-f8a3861cddfa.png)

**Comment by palob** [comment]
But we haven't got any leverage on what the user chooses there so I think we need to fix the ``index.theme``

**Comment by wa4557** [comment]
I agree, for me this seems to be a Qt issue, and needs to be fixed with the `index.theme` entries...

**Comment by hlechner** [comment]
Yea, defiantly we need to find a solution/workaround for this.

the easier and maybe better option is to change it to **Fixed**.

But I think it should be also reported to the developers even if we make some workaround, because they probably didn't test this with a icon theme without **Fixed** sizes. Being honest, it's not only the numix that uses this way.

**Comment by palob** [comment]
But from the ones I tested only Numix exhibits the issue.

Regarding your example above that's Qt4 which will be EOL in a week's time - meaning no future development whatsoever.

**Comment by hlechner** [comment]
Interesting I didn't know about EOL of qt4. I think the **Fixed** sizes of the small icons: 16,22,24 would avoid bad programming in this sense even in pure GTK+, imagine a developer who define a size of **21px**, it could be resized from 22 but t will be a sh*t. instead of this is better put a 16px fixed on that.

Maybe for the higher sizes would't be a big issue like the small ones.

**Comment by palob** [comment]
Even when applying the fix (all type of 16/22/24/32 and respective x2 icons to ``Type=Fixed`` save for ``16/emblems``) the GTK2 filechooser still shows this:
![blurry2](https://cloud.githubusercontent.com/assets/7050624/11768908/4ed1e0fe-a1db-11e5-963e-dd318c56272d.png)

Octave still shows this:
![blurry-octave](https://cloud.githubusercontent.com/assets/7050624/11768911/58a4d122-a1db-11e5-851e-1fa63142834d.png)

That's quite odd. It's the 48px icon showing up instead of the 24px ,the 64px icon showing up instead of the 32px one and whatever one instead of the 16px one (you can change the icon size in Octave).

Maybe that's due to how our ``places`` (and related icons in ``actions``) icons are designed?
They have got a shadow which exceeds the canvas, for instance a width of 48px in the 32px icon.

**Comment by wa4557** [comment]
Why should the places icons interfere with action icons? Strange...

**Comment by palob** [comment]
Well, what I wanted to say is that the action icons like ``document-open`` are designed with a shadow underneath like ``places`` icons.

**Comment by wa4557** [comment]
Maybe it's the `MinSize` variable for 48 and 64?

**Comment by palob** [comment]
Changed the values to 48 and 64 respectively for ``actions`` and ``places`` but it doesn't seem to be of any avail.

**Comment by bil-elmoussaoui** [comment]
@palob Do you update the icon cache when you do modify the index file?

**Comment by wa4557** [comment]
ut we need the `scalable` part for these sizes :(

**Comment by palob** [comment]
I manually removed the cache file and cleared the thumbnail cache.

**Comment by palob** [comment]
@wa4557 Changing ``48`` and ``64`` to ``Fixed`` doesn't do the trick anyway. 
Might be an Octave bug then.

**Comment by wa4557** [comment]
And the icons in VLC show up correctly now?

**Comment by palob** [comment]
Qt filechooser, VLC and all other Octave icons are OK.

**Comment by hlechner** [comment]
Most of my apps are GTK or other GUIs rather than QT, but you guys have found this issue on QT5 apps too?

**Comment by palob** [comment]
A screenshot of Dolphin 15.12 from the Arc repo used in GNOME 3.18:
![dolphin](https://cloud.githubusercontent.com/assets/7050624/12467455/c89fb352-bfdb-11e5-81b0-3f4996fad9d6.png)

Blurry icons. However on Kubuntu 15.10 (booted into live iso) the icons are crisp.
Is it the Arch repo or does KDE/Qt behave differently in this regard when there is GTK? Mutter? present?

**Comment by bil-elmoussaoui** [comment]
I think this bug is fixed after merging the PR from @wa4557 
A screenshot from Octave
![capture du 2016-04-03 00-11-46](https://cloud.githubusercontent.com/assets/7660997/14229306/d9d4f428-f930-11e5-91f9-5b1b89b57e4c.png)
![capture du 2016-04-03 00-12-05](https://cloud.githubusercontent.com/assets/7660997/14229308/db08a81c-f930-11e5-93ec-b6ade2010a76.png)

**Comment by palob** [comment]
It is indeed :+1:
